### DISCUSSION TITLE
clicking "commit" too soon is not handled

### DISCUSSION THREAD

**Discussion by DrCray** [community/community#57058] [New to GitHub]
When uploading a file to a project/pages, if you hit the "commit" button before the file is finished uploading, you get a blank page. Solution: make commit button inactive til all uploads finish.  For newbies, like me, it quite freaked me out. Also, when that happens, the commit doesn't take, so I'm blundering around looking for my file.  I got it now, but this idea could help other newbies, especially those of us who are a bit "tech-challenged". Thanks

**Comment by neutrinomuon** [comment]
Hi @DrCray,

I really appreciate your suggestion; it's a great idea :)! While it ultimately depends on GitHub's decision and timeline for implementing such a feature, there are at least two helpful workarounds that can help users with the issue you just mentioned:

1. Use a Git client: Instead of relying solely on the web interface, consider utilizing a Git client such as Git command-line tools or a Git desktop client like [GitHub Desktop](https://desktop.github.com/) or [Sourcetree](https://www.sourcetreeapp.com/). These tools offer more control over the commit process, allowing you to stage and commit changes locally before pushing them to GitHub. This way, you can ensure that all files have been successfully uploaded before initiating the commit.

2. Create a new branch in your repository: Instead of committing directly to the main branch, consider creating a separate branch for your changes and tests. This approach enables you to make multiple commits and refine your changes without affecting the main branch. Once the alternative branch is working well, then the files are ready to go, and you can merge the branch into the main branch.

I hope these are useful suggestions for other users facing the same problem.

**Reply by DrCray** [reply]
My own experience with the new generation, is that they prefer the direct web interface rather than the client or command line terminal. This means they can access anything anywhere at any time on any device. 
As Github has expanded with it's pages that can be used for any kind of project--such as mine which is educational, not technical--this is just a suggestion as I'm suspecting Github would want to adjust to the new target market. Nevertheless,  I appreciate your suggestions and your taking the time to do it.
